intellij/src/saros/intellij/ui/util/SafeDialogUtils.java

```diff
@@ -193,4 +194,34 @@ public static Integer showYesNoDialog(Project project, final String message, fin
 
     return response.get();
   }
+
+  /**
+   * Shows a non-blocking yes/no dialog. The passed <code>runnable</code> can be used to run code
+   * after the dialog is finished. However, it is <b>only</b> run if the user finishes the dialog
+   * with the option {@link Messages#YES}.
+   *
+   * @param project the project used as a reference to generate and position the dialog
+   * @param message the text displayed as the message of the dialog
+   * @param title the text displayed as the title of the dialog
+   * @param runAfter the runnable to execute if the user chooses {@link Messages#YES}
+   */
+  public static void showYesNoDialog(
+      @NotNull Project project,
+      @NotNull String message,
+      @NotNull String title,
+      @NotNull Runnable runAfter) {
+
+    LOG.info("Showing non-blocking yes/no dialog: " + title + " - " + message);
```

Comment:
debug ?

Reply:
The info log level is currently used for all dialog methods. I am not sure what exactly led me to make this decision. We previously had a guideline when to use the different log-levels, but it got dropped during the website migration. (It was not that helpful anyways.)

I think it was something along the lines that we would like to know what was shown to the user when reading logs of released plugin versions (as released versions only log info and above). They can, for example, be used to get some information about the state of the session and session actions. For this reason, any dialog and notification is currently logged with the log level `info`.

Currently, I am not sure whether the release logging restriction even applies for Saros/I (I don't think so), but it still doesn't hurt to have these as an indication of the plugin state. This is especially useful for me as I use a plugin to fold any debug and trace loggin by default in the IDE, which allows me to only see the important information by default but still leaves me the option to check the debug loggin in the IDE when needed.

Either way, if this should get changed to debug, I would also change it for the other methods and the notification class (and move this change to a separate commit). But currently, I am leaning towards leaving it as `info` logging.